Application/Services/CharacterServices/CharacterServices.cs

```diff
@@ -0,0 +1,248 @@
+﻿using Application.Interfaces;
+using Domain.Entities.Character;
+using System;
+using System.Collections.ObjectModel;
+
+namespace Application.Services.CharacterServices;
+
+public class CharacterServices : ICharacterServices
+{
+    public readonly IGenericRepository<CharacterAsset> _characterAssetRepo;
+    public readonly IGenericRepository<CharacterAttribute> _characterAttributeRepo;
+    public readonly IGenericRepository<CharacterEntity> _characterRepo;
+    public readonly IGenericRepository<CharacterType> _characterTypeRepo;
+
+    public CharacterServices(IGenericRepository<CharacterAsset> characterAssetRepo
+        , IGenericRepository<CharacterAttribute> characterAttributeRepo
+        , IGenericRepository<CharacterEntity> characterRepo
+        , IGenericRepository<CharacterType> characterTypeRepo)
+    {
+        _characterAssetRepo = characterAssetRepo;
+        _characterAttributeRepo = characterAttributeRepo;
+        _characterRepo = characterRepo;
+        _characterTypeRepo = characterTypeRepo;
+    }
+
+    //Character Asset
+    public async Task<ICollection<CharacterAsset>> GetCharacterAssets()
+    {
+        var charAss = await _characterAssetRepo.ListAsync();
+        return charAss;
+    }
+    public async Task<CharacterAsset> GetCharacterAsset(Guid characterAssetId)
+    {
+        var charAss = await _characterAssetRepo.FindByIdAsync(characterAssetId);
+        if (charAss == null)
+        {
+            throw new Exception($"Character asset not exist");
+        }
+        else
+        {
+            return charAss;
+        }
+    }
+    public async Task<ICollection<CharacterAsset>> GetCharacterAssets(Guid id, int typeId) { // Type ID: 1: AssetId, 2: CharacterId
+        ICollection<CharacterAsset> charAss = new Collection<CharacterAsset>();
+        if(typeId == 1)
+        {
+            charAss = await _characterAssetRepo.WhereAsync(
+                cA => cA.AssetsId.Equals(id));
+        } else if (typeId == 2)
+        {
+            charAss = await _characterAssetRepo.WhereAsync(
+                cA => cA.CharacterId.Equals(id));
+        }
+        if(charAss.Count == 0)
+        {
+            throw new Exception($"Character Asset or Asset Id / CharacterId not found");
+        }
+        else
+        {
+            return charAss;
+        }
+    }
+    public async Task<int> CountCharacterAssets()
+    {
+        var charAss = await _characterAssetRepo.ListAsync();
```

Comment:
we already have SumAsync function in Repository, why we need this implementation?